### PR TITLE
Add map strategy

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add map strategy to replace the map panel.
No options for now, some can come in the future.

### Create a dashboard dialog
CleanShot 2024-03-12 at 11 04 08

### Map dashboard
CleanShot 2024-03-12 at 11 02 37

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by Misiu** [comment]
@piitaya why don't we do the opposite and select the people and devices we want to see on the map?
My argument is quite simple, I recently added [Huawei LTE](https://www.home-assistant.io/integrations/huawei_lte/) integration and now I have around 60+ device trackers for all the wifi connected devices that I have in my home, so instead of hiding 60 devices I would prefer to select only those devices and people that I want to see on the map.

This is my personal preference and I'm sure there was a reason for the option you picked to implement, but I'd like to understand it.

**Comment by karwosts** [comment]
The old map panel had the pencil to open the zone editor, is that no longer going to exist?

File: src/panels/lovelace/strategies/map/map-view-strategy.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import { computeStateDomain } from "../../../../common/entity/compute_state_domain";
+import { LovelaceViewConfig } from "../../../../data/lovelace/config/view";
+import { HomeAssistant } from "../../../../types";
+import { MapCardConfig } from "../../cards/types";
+
+export type MapViewStrategyConfig = {
+  type: "map";
+};
+
+export const getMapEntities = (hass: HomeAssistant) => {
```
Comment (bramkragten) [inline src/panels/lovelace/strategies/map/map-view-strategy.ts:12]: Why is this exported?

**Review by bramkragten — DISMISSED** [review]

File: src/panels/lovelace/strategies/map/map-view-strategy.ts (hunk 1 above)
Comment (piitaya) [reply src/panels/lovelace/strategies/map/map-view-strategy.ts:12]: It was used in the editor but it's not used anymore

**Comment by piitaya** [comment]
> The old map panel had the pencil to open the zone editor, is that no longer going to exist?

Yes, it's removed. In dashboard  pages, the edit button is used to edit the dashboard so it was confusing to have a edit button to edit zones.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**